frontend/test/account-send.test.ts

```diff
@@ -0,0 +1,561 @@
+      encryptionCount: BigNumber.from(2).pow(255).toString(),
+      expectedDecryptedData: {
+        advancedMode: false,
+        usePublicKeyChecked: true,
+        address: recipientAddress,
+        pubKey: partialPubKey,
+      },
+    },
+  ].forEach(({ ciphertext, encryptionCount, expectedDecryptedData }) => {
+    it('Data decrypted correctly all true', () => {
+      const data = decryptData(ciphertext, { encryptionCount, viewingPrivateKey: viewingPrivateKey });
+      expect(data).toEqual(expectedDecryptedData); // Recursively checks all properties.
+    });
+  });
+});
+
+describe('encrypt/decrypt relationship', () => {
+  it('Encrypt and decrypt are inverses', () => {
+    // Generate random data and do multiple runs.
+    for (let i = 0; i < 100; i++) {
+      const { recipientAddress, pubKey, viewingPrivateKey, advancedMode, usePublicKeyChecked, encryptionCount } =
+        randomData();
+      const ciphertext = encryptAccountData(
+        { recipientAddress, advancedMode, pubKey, usePublicKeyChecked },
+        { encryptionCount, viewingPrivateKey }
+      );
+      const decryptedData = decryptData(ciphertext, { encryptionCount, viewingPrivateKey });
+      expect(decryptedData).toEqual({
+        advancedMode,
+        usePublicKeyChecked,
+        address: recipientAddress,
+        pubKey: `0x99${pubKey.slice(4, 4 + 22)}`,
+      });
+    }
+  });
+
+  it('Encrypts and decrypts properly for values with leading zeros', () => {
+    // `BigNumber.from(myHexString).toHexString()` where `myHexString` has leading zero bytes will
+    // truncate those leading zeroes from the string. If data is not manually zero-padded to the
+    // correct length, this can cause a bug where the leading zero bytes of the address are
+    // truncated and trailing bytes of other data are appended. The above fuzz test found this bug,
+    // and this is a concrete test with known inputs would fail if the bug were present.
+    const recipientAddress = '0x00bB833415cf56f112389E146Cb7847dDbE93fB5';
+    const pubKey =
+      '0x040426c84897f9e07632687f69ef714090307dcdc23b4a9f3719e6db8eb1f753f71dd4f849c6101d13049f6091944914c31e6fe32db34daa176da230a04de12bf3';
+    const viewingPrivateKey = '0xeaa492b979aead8bdaf857673c4fe453a3f7ce5e8d661499391505f47d96d613';
+    const advancedMode = false;
+    const usePublicKeyChecked = false;
+    const encryptionCount = '661';
+
+    const ciphertext = encryptAccountData(
+      { recipientAddress, advancedMode, pubKey, usePublicKeyChecked },
+      { encryptionCount, viewingPrivateKey }
+    );
+    const decryptedData = decryptData(ciphertext, { encryptionCount, viewingPrivateKey });
+    expect(decryptedData).toEqual({
+      advancedMode,
+      usePublicKeyChecked,
+      address: recipientAddress,
+      pubKey: `0x99${pubKey.slice(4, 4 + 22)}`,
+    });
+  });
+});
+
+describe('storeSend', () => {
+  const testingUtils = generateTestingUtils({ providerType: 'default' });
+  const storeSendArgs = {
```

Comment:
Can we genreate random data for all these tests like we do above? Using random data has already found us one bug, so we should avoid hardcoding data for this too to write stronger tests

Reply:
Yes, good point will do

Reply:
Using the random data generation https://github.com/ScopeLift/umbra-protocol/pull/494/commits/23e95594444466c2dd65e2834dfe95ed665f79db

Reply:
Ah good call will update

Reply:
https://github.com/ScopeLift/umbra-protocol/pull/494/commits/28153b80deb22ea644bb2085519cc5e5f776669f